### DISCUSSION TITLE
Want to update a single table column with data from another table & column.

### DISCUSSION THREAD

**Discussion by mbriggs9374** [crate/crate#10642] [Q&A] [answered]
I have two tables (data1 & data2), data1 only needs column1 to be updated. So I tried this:
```
update d1
set d1.location = d2.location
from data1 d1
inner join data2 d2 on d1.id = d2.id;
```
That didn't work so I then tried to do this:
```
update data1 as d1
set location = (select location from data2 d2 where d1.id = d2.id);
```
That produced this Error: `SQLActionException[UnsupportedFeatureException: Subquery not supported in this statement]`

I'm open to just about any ideas and/or suggestions.

**Comment by mfussenegger** [comment, marked as answer]
Update operations on joins or correlated subqueries are currently not supported.

There are issues about that:

- https://github.com/crate/crate/issues/8132
- https://github.com/crate/crate/issues/9320

There's currently no good workaround, other than selecting the ids and then generating an update statement on the client side.
